demo.js

```diff
@@ -8,11 +8,29 @@ fluid.defaults("flock.demo.oscilOscil", {
 
         playButton: {
             type: "flock.ui.enviroPlayButton",
-            container: "{oscilOscil}.container"
+            container: "{oscilOscil}.container",
+            options: {
+                listeners: {
+                    // Due to a bug in Flocking, we need to
+                    // connect/disconnect the Nexus components
+                    // whenever the environment is played or pauses.
+                    "onPlay.connectVisualization": {
+                        func: "{nexusui}.events.onConnect.fire"
+                    },
+                    "onPause.disconnectVisualization": {
+                        func: "{nexusui}.events.onDisconnect.fire"
+                    }
+                }
+            }
         },
 
         synth: {
             type: "flock.demo.oscilOscil.synth"
+        },
+
+        nexusui: {
+            type: "flock.demo.oscilOscil.nexusui",
+            container: ".viz"
```

Comment:
Rather than hard coding this, why not make the oscilOscil demo a view component, and give it a root container (perhaps a ```<main>`` element wrapping both the visualization elements and the play button? Then it's a fully self-contained view, and you can configure the visualization view's container in the ```selectors``` block.

Reply:
I've used divs and spans, since these seem to work with the way that NexusUI expects to see the DOM formatting.

Reply:
Sorry, I  wasn't clear. Typically one uses only  one ```<main>``` element in a document, and it communicates to the user agent and assistive technologies where the [essential portion of the page's UI is](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/main), which is why I suggested using that tag for your ```#oscilOscil``` wrapper element.

Nexus UI shouldn't care what the parent element is, as long as it gets divs for its own elements—your  ```#scope``` and ```#spectrogram``` tags, which you do.

But <div>s are just fine, doesn't matter.